Feat/ev 1175/node02   introdução typescript

PaulinMarcio opened the pull request:
NodeJS - Card 02: Introdução ao Typescript

Objetivos:
Deverá ser instalado e configurado o typescript  - Feito
Deverá ser instalado e configurado também o ESLINT - Feito
Os argumentos recebidos nas funções deverá conter uma tipagem - Feito
As funções e métodos deverão ter um retorno definido - Feito
Utilizar o mínimo possível a tipagem “any” - Feito
As variáveis ou constantes que irão armazenar objetos ou dados deverão ser tipadas em caso onde um dado de um tipo necessite ser tipado com outro tipo. - Feito

PaulinMarcio committed:
Feat: WIP transição para TS

PaulinMarcio committed:
feat wip validation

PaulinMarcio committed:
wip

PaulinMarcio committed:
feat: conversão da api para typescript

PaulinMarcio committed:
Feat: tipagem restante

tandy-souza commented:
Ainda tem muitos arquivos js, não seria para converter todos para ts? Será feito posteriormente?

ismaelalvesgit commented:
a pasta `dist` não deve ficar no PR aberto por favor adicione no gitignore a pasta e remova a pasta do PR

ismaelalvesgit commented on src/domain/user/services/HandleValid.ts:
```diff
@@ -0,0 +1,16 @@
+import { userSchema } from "./UserValid";
+import users from "../mocks/UserMock"
+
+const validation = (schema: typeof userSchema) => async(req: Request, res:Response, next:NextFunction) => {
```
Acho que assim ficaria melhor

```ts
export const validation = (schema: typeof userSchema) => async(req: Request, res:Response, next:NextFunction)
```

ismaelalvesgit commented on src/domain/user/services/UserValid.ts:
```diff
@@ -0,0 +1,24 @@
+import { object, string, date, number, boolean, ref } from 'yup';
+
+const userSchema = object().shape({
```
Acho que assim ficaria melhor

```ts
export const userSchema = object().shape({
```

ismaelalvesgit commented on src/presentation/controllers/create/createUser.ts:
```diff
@@ -0,0 +1,13 @@
+import users from '../../../domain/user/mocks/UserMock';
```
Sugiro que no futuro comesse a utilizar [path-mapping](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping) [aqui](https://blog.rocketseat.com.br/path-mapping-typescript/) um exemplo de como fazer.

ismaelalvesgit commented on src/presentation/controllers/create/createUser.ts:
```diff
@@ -0,0 +1,13 @@
+import users from '../../../domain/user/mocks/UserMock';
+import {Request, Response} from "express"
+
+class createUser {
```
Acho que assim ficaria melhor

```ts
export class createUser 
```

ismaelalvesgit commented on src/presentation/controllers/list/listUser.ts:
```diff
@@ -0,0 +1,11 @@
+import userData from "../../../domain/user/mocks/UserMock";
+import { Request, Response } from "express";
+
+class allUsers{
```
Acho que assim ficaria melhor

```ts
export class allUsers
```

PaulinMarcio replied:
Feito

PaulinMarcio committed:
Feat: Correção dos sherpas

Correção das revisões dos sherpas:

Remoção dos arquivos JS,
Atualização dos exports,
Utilização do path mapping

PaulinMarcio committed:
Update createUser.ts

ismaelalvesgit commented on package.json:
```diff
@@ -2,16 +2,25 @@
   "license": "MIT",
   "dependencies": {
     "express": "^4.17.3",
+    "tsconfig-paths": "^4.1.2",
```
essa lib como e utilizada somente em tempo de desenvolvimento então deveria estar em devDependencies

PaulinMarcio replied:
Feito

PaulinMarcio committed:
conversão tsconfig-paths de dep para dev-dep
